vault-tool/src/Network/VaultTool/KeyValueV2.hs

```diff
@@ -0,0 +1,237 @@
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+{- | A library for working with Vault's KeyValue version 2 secrets engine
+
+ Unless otherwise specified, all IO functions in this module may
+ potentially throw 'HttpException' or 'VaultException'
+-}
+module Network.VaultTool.KeyValueV2 (
+    VaultSecretVersion (..),
+    VaultSecretVersionMetadata (..),
+    VaultSecretMetadata (..),
+    vaultWrite,
+    vaultRead,
+    vaultReadVersion,
+    vaultDelete,
+    vaultList,
+    isFolder,
+    vaultListRecursive,
+) where
+
+import Control.Applicative ((<|>))
+import Data.Aeson (
+    FromJSON,
+    ToJSON,
+    object,
+    parseJSON,
+    toJSON,
+    withObject,
+    (.:),
+    (.=),
+ )
+import Data.Text (Text)
+import qualified Data.Text as T
+import Data.Time (UTCTime)
+
+import Network.VaultTool.Internal (
+    newDeleteRequest,
+    newGetRequest,
+    newListRequest,
+    newPostRequest,
+    runVaultRequestAuthenticated,
+    runVaultRequestAuthenticated_,
+    withStatusCodes,
+ )
+import Network.VaultTool.Types (
+    VaultConnection,
+    Authenticated,
+    VaultMountedPath (..),
+    VaultSearchPath (..),
+    VaultSecretPath (..),
+ )
+
+data VaultSecretMetadata = VaultSecretMetadata
+    { _VaultSecretMetadata_leaseDuration :: Int
+    , _VaultSecretMetadata_leaseId :: Text
+    , _VauleSecretMetadata_renewable :: Bool
+    }
+    deriving (Show, Eq {- TODO Ord -})
```

Comment:
What is preventing this TODO?

Reply:
We're not actually using this metadata type right now so I'm removing it.

Reply:
2c42064fa561f294710ccd8b3ca290ace3d42d3f